### PR TITLE
Fallback Encoding of CStore service (#1159)

### LINKED ISSUE #1159: FallBackEncoding not applied on C-Find request when no SpecificCharacterSet tag is present
When no SpecificCharacterSet tag is provided in C-Find request to foDicom based SCP, foDicom seems to use default charset (i.e. ASCII) to decode e.g. PN tags, which results in '?' in place of certain chars.  After a quick look at the source, 
was it intentional for any reason to use DicomDatasetReaderObserver contructor with only a command (DicomService->ProcessPDataTFAsync) instead of providing also the fallBackEncoding available in DicomService?
e.g.
`var reader = new DicomReader();
                            reader.IsExplicitVR = false;
                            reader.Read(new StreamByteSource(_dimseStream, FileReadOption.Default), new DicomDatasetReaderObserver(command, _fallbackEncoding));`
instead of:
`var reader = new DicomReader();
                            reader.IsExplicitVR = false;
                            reader.Read(new StreamByteSource(_dimseStream, FileReadOption.Default), new DicomDatasetReaderObserver(command));`

Best regards
Jarek

Fellow Oak DICOM 4.0.7
OS: Windows 10 x64
Platform: NET Framework 4.5.2

### PR DESCRIPTION
**Pull request by bjorn-malmo**
Fixes #1159.

The fallback encoding specified when creating a service from DicomServerFactory should be applied to downstream objects.

This issue is not resolved in the repo, and this poses a problem when interacting with non-conformant modalitites, such as Olympus EndoAlpha (see excerpt from Olympus DCS).

olympus-dcs

#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
- [ ] I have updated API documentation
- [x] I have included unit tests
- [x] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file

### CODE REVIEW

File: FO-DICOM.Core/DicomTransferSyntax.cs [hunk 1]
```diff
@@ -713,12 +713,19 @@ static DicomTransferSyntax()
...
+            Entries.Add(FragmentableMPEG4AVCH264HighProfileLevel41.UID, FragmentableMPEG4AVCH264HighProfileLevel41);
             Entries.Add(MPEG4AVCH264BDCompatibleHighProfileLevel41.UID, MPEG4AVCH264BDCompatibleHighProfileLevel41);
+            Entries.Add(FragmentableMPEG4AVCH264BDCompatibleHighProfileLevel41.UID, FragmentableMPEG4AVCH264BDCompatibleHighProfileLevel41);
             Entries.Add(MPEG4AVCH264HighProfileLevel42For2DVideo.UID, MPEG4AVCH264HighProfileLevel42For2DVideo);
+            Entries.Add(FragmentableMPEG4AVCH264HighProfileLevel42For2DVideo.UID, FragmentableMPEG4AVCH264HighProfileLevel42For2DVideo);
             Entries.Add(MPEG4AVCH264HighProfileLevel42For3DVideo.UID, MPEG4AVCH264HighProfileLevel42For3DVideo);
+            Entries.Add(FragmentableMPEG4AVCH264HighProfileLevel42For3DVideo.UID, FragmentableMPEG4AVCH264HighProfileLevel42For3DVideo);
             Entries.Add(MPEG4AVCH264StereoHighProfileLevel42.UID, MPEG4AVCH264StereoHighProfileLevel42);
+            Entries.Add(FragmentableMPEG4AVCH264StereoHighProfileLevel42.UID, FragmentableMPEG4AVCH264StereoHighProfileLevel42);
```
Comment (mrbean-bremen) [inline FO-DICOM.Core/DicomTransferSyntax.cs:728]: These seem to belong to another issue

**Review by mrbean-bremen — APPROVED** [review]

**Review by gofal — APPROVED** [review]

**Comment by ayedo** [comment]
Do you think this should solve this problem too #1642  ?

**Comment by bjorn-malmo** [comment]
Yes,

I can confirm that CFind requests will use the FallbackEncoding, if it is supplied in the constructor of the CStoreRequest.

Our implementation looks like this:

    public class CStoreRequest : DicomService, IDicomServiceProvider, IDicomCStoreProvider, IDicomCEchoProvider
    {

        public CStoreRequest(INetworkStream stream, Encoding fallbackEncoding, ILogger log, DicomServiceDependencies dependencies)
            : base(stream, CStoreProvider.Singleton.FallbackEncoding, log, dependencies)
        {
        }

   [snip]

 Regards Björn



________________________________

**Comment by ayedo** [comment]
Thanks for your reply! I'm not sure I understand, our problem was on the SCU side, when we send a C-Find request, and get replies from a PACS that does not include the SpecificCharacterSet, the fallback encoding would not work. The code you showed seems to be on the SCP side?

**Comment by amoerie** [comment]
@bjorn-malmo Changes look good to me, but could you remove the added transfer syntaxes? They seem accidental?

**Comment by mrbean-bremen** [comment]
This should be rebased against development and can be merged afterwards (moving the change log entry up to 5.1.3). The extra changes mentioned in the review have been there due to incorrect rebasing, so all should be ok (I would do the rebasing, but don't have the rights).

**Pull request closed without merging**